good-enough-practices-for-scientific-computing.tex

```diff
@@ -176,102 +172,55 @@ \section{Data Management}\label{sec:data}
-  The best way to do this is to write scripts for
-  \emph{every} stage of data processing.
-  This not only ensures that you have a record, but also allows you to
-  re-do the cleanup and other steps when new data arrives.
-  In cases where automated data processing is not possible
-  or not practical, it's important to
-  clearly document every manual action taken (what menu was used, what
-  column was copied and pasted, what link was clicked, etc.). For example,
-  choosing a region of interest in an image
-  is inherently interactive, but you can save the region chosen
-  as a set of boundary coordinates. If the
-  exact details of the procedure cannot easily be recorded,
-  save intermediate results for later checking.
+  \emph{Make each row an observation}: Data often comes in a wide format, because that
+  facilitated data entry or human inspection. Imagine one row per field site and then
+  columns for measurements made at each of several time points. Be prepared to gather such
+  columns into a variable of measurements, plus a new variable for time point.
 
 \item
-  \recommend{Give every record a unique key}.  In a well-organized
-  data set, it's easy to precisely refer to a specific record.
-  Using fields in the data (such as people's names or addresses) does
-  not guarantee this; using row numbers in tables does, but those
-  numbers will change as records are added, removed, or sorted.
-
-  We therefore recommend giving each record a unique, persistent key
-  that has no other purpose than identifying the
-  record\footnote{Student numbers and similar identifiers serve this
-    purpose in institutional databases.} The provision of such a key
-  also makes it much easier to link data sets together.
+  \recommend{Record all the steps used to process data}: Data manipulation is as integral
+  to your analysis as statistical modelling and inference. If you do not document this
+  step thoroughly, it impossible for you, or anyone else, to repeat the analysis.
+  
+  The best way to do this is to write scripts for \emph{every} stage of data processing.
+  This might feel frustratingly slow, but you will get faster with practice. The
+  immediate payoff will be the ease with which you can re-do data preparation when new
+  data arrives. You can also re-use data preparation logic in the future for related
+  projects.
+
+  When scripting is not feasible, it's important to clearly document every manual action
+  (what menu was used, what column was copied and pasted, what link was clicked, etc.).
+  Often you can at least capture \emph{what} action was taken, if not the complete
+  \emph{why}. For example, choosing a region of interest in an image is inherently
+  interactive, but you can save the region chosen as a set of boundary coordinates.
 
 \item
-  \recommend{Include complementary data in additional files}.  Raw data
-  frequently does not fully explain
-  itself. For example, if each well in an $8{\times}12$ microtitre
```

Comment:
@lexnederbragt You never liked my micro titre plates 😜. At least, I think it was you. How you like me now? New example is about humans and heart rate.